source/brailleDisplayDrivers/albatross/__init__.py

```diff
@@ -0,0 +1,677 @@
+		self._invalidSettingsByte = False
+		# Set to True when filtering redundant init packets when init byte
+		# received but not yet settings byte.
+		self._initByteReceived = False
+		# Set to True if settings byte has not been dequeued yet in
+		# _handleInitPackets.
+		self._waitingSettingsByte = False
+		# Set to True if ctrl key combination is partially dequeued.
+		self._waitingCtrlPacket = False
+		# Port object
+		self._dev = None
+		# Try to connect or reconnect
+		self._tryToConnect = False
+		self._readQueue = deque()
+		self._writeQueue = deque(maxlen=WRITE_QUEUE_LENGTH)
+		self._exitEvent = Event()
+		# Thread for read
+		self._handleRead = None
+		self._writeLock = Lock()
+		# Timer to keep connection (see KC_INTERVAL).
+		self._kc = None
+		# When previous write was done (see KC_INTERVAL).
+		self._writeTime = 0.0
+		# Search ports where display can be connected.
+		for portType, portId, port, portInfo in self._getTryPorts(port):
+			# For reconnection
+			self._currentPort = port
+			self._tryToConnect = True
+			self._readHandling()
+			if self.numCells:
+				# Prepare _oldCells to store last displayed content.
+				while len(self._oldCells) < self.numCells:
+					self._oldCells.append(0)
+				self._kc = threads.RepeatedTimer(
+					KC_INTERVAL,
+					self._keepConnected
+				)
+				self._handleRead = threads.ReadThread(
+					self._readHandling,
+					self._disableConnection,
+					self._exitEvent,
+					self._dev,
+					name="albatross_read",
+					daemon=True
+				)
+				self._handleRead.start()
+				log.info(
+					f"Connected to Caiku Albatross {self.numCells} on {portType} port {port} "
+					f"at {BAUD_RATE} bps."
+				)
+				break
+			# This device initialization failed.
+			if self._dev:
+				if self._dev.is_open:
+					self._dev.close()
+				self._dev = None
+			if self._invalidSettingsByte:
+				log.info(
+					"After checking internal settings, switch display off and on, "
+					"and if needed restart NVDA"
+				)
+			log.info(
+				f"Connection to Caiku Albatross display on {portType} port {port} "
+				f"at {BAUD_RATE} bps failed."
+			)
+		else:
+			raise RuntimeError("No Albatross found")
+
+	def terminate(self):
+		try:
+			if self._handleRead:
+				self._exitEvent.set()
+			super().terminate()
+			self.numCells = 0
+			if self._dev:
+				if self._dev.is_open:
+					self._dev.close()
+				self._dev = None
+			if self._kc:
+				self._kc.stop()
+				self._kc = None
+			if self._readQueue:
+				self._readQueue.clear()
+			if self._writeQueue:
+				self._writeQueue.clear()
+			self._readQueue = None
+			self._writeQueue = None
+			self._handleRead = None
+		except Exception:
+			# Terminating anyway
+			pass
+
+	# _initConnection, _initPort, _openPort, _readInitByte and _readSettingsByte
+	# are helper functions to establish connection.
+	# If no connection, Albatross sends continuously INIT_START_BYTE
+	# followed by byte containing various settings like number of cells.
+	def _initConnection(self) -> bool:

```

Comment:
Comments like these should be docstrings.

Can you also add an explanation on why this must sleep

```suggestion
	def _initConnection(self) -> bool:
		"""
		_initConnection, _initPort, _openPort, _readInitByte and _readSettingsByte
		are helper functions to establish connection.
		If no connection, Albatross sends continuously INIT_START_BYTE
		followed by byte containing various settings like number of cells.
		"""
```